Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -28,16 +28,31 @@ module.exports.Quantity = class Quantity extends Expression
     if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit 
+      @value < parseFloat other.value 
+    else if other instanceof Quantity and other.unit of time_units and @unit of time_units
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
     if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
 
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+time_units = {'years': 'year', 'yr': 'year', 'y': 'year', 'months': 'month', 'weeks': 'week', 'wk': 'week', 'wks': 'week', 'days': 'day', 'd': 'day', 'minutes': 'minute', 'min': 'minute', 'seconds':'second', 'sec':'second', 'second':'second', 'milliseconds' : 'millisecond' }
```

Comment:
I'm pretty sure `yr`, `y`, and `wks` aren't valid time units in CQL or UCUM.  Some of the new ones above are UCUM, which is probably fine -- but if you're going to add UCUM, you might as well add them all.  See the "Time Units" section here: http://download.hl7.de/documents/ucum/ucumdata.html